pkgs/tools/filesystems/fuse-archive/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, pkgs
```

Comment:
**Do not use** `pkgs` as a parameter.

```suggestion
```

Reply:
Just so I know for the future - what's the reason for not using `pkgs`? If you have a link I can read more about it, please let me know.

Reply:
There is not a written reason for this. But I can think in a few:

1. `pkgs` encompass all the packages of Nixpkgs, indirectly bringing them to the scope of your expression. Although Nix evaluates things lazily, it is not an elegant thing to do.
2. An expression shall be "self explanatory". Pulling things from `pkgs` is not clean. 
3. It makes more difficult to customize the function.

   I can give an example I bumped before knowing NixOS. IIRC a long time ago a certain version of [qemu](https://www.qemu.org/) employed a very specific set of undocumented GCC extensions in its code.
Because of it, when the distro I used at that time upgraded GCC to a major version (removing the older one), qemu could not be built (because this new GCC removed that extensions).

   In Nixpkgs this problem can be easily solved by a code like this:

   ```nix
       qemu = callPackage ./qemu.nix { stdenv = olderGccStdenv; };  
   ```

   However, this is possible only if `stdenv` is an explicit parameter of qemu.nix:

   ```nix
      { lib, stdenv, fetchurl, pka, pkb, pkc, . . . }
   ```

   If you "hide" stdenv under `pkgs.stdenv`, then it is impossible to "downgrade" it.

Reply:
Got it, thanks for a good explanation! I'll try to keep this in mind.